gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagProcFactory.java

```diff
@@ -48,41 +53,52 @@
 @Singleton
 public class DagProcFactory implements DagTaskVisitor<DagProc<?>> {
 
+  private final Config config;
   private final FlowCompilationValidationHelper flowCompilationValidationHelper;
+  private final List<Class<? extends Exception>> nonRetryableExceptions;
 
   @Inject
-  public DagProcFactory(FlowCompilationValidationHelper flowCompilationValidationHelper) {
+  public DagProcFactory(Config config, FlowCompilationValidationHelper flowCompilationValidationHelper) {
+    this.config = config;
     this.flowCompilationValidationHelper = flowCompilationValidationHelper;
+    this.nonRetryableExceptions = ConfigUtils.getStringList(config, ServiceConfigKeys.DAG_PROC_ENGINE_NON_RETRYABLE_EXCEPTIONS_KEY)
```

Comment:
this doesn't appear used anywhere.  did you meant to pass it along to the `DagProc`s this creates?  that would better than providing them the full config.

even better than giving them a `List<Class<? extends Exception>>` would be to provide each `DagProc` ctor a `Function<Throwable, Boolean>` predicate.

Reply:
ah, missed removing it during refactor

Reply:
if i have to pass an argument, i would like to pass something that can be used for any future requirements, this way param counts will remain in check.
i would prefer to pass `Config` instead of `Predicate`